src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -0,0 +1,240 @@
+
+        section2 = new AttendanceSection();
+        section2.setSectionId(SECTION_2_ID);
+        section2.setCanvasCourseId(COURSE_ID);
+        section2.setCanvasSectionId(SECTION_2_ID);
+        section2.setName(SECTION_2_NAME);
+
+        studentCommentsMap1 = new HashMap<>();
+        studentCommentsMap1.put(STUDENT_1_ID, STUDENT_1_COMMENT);
+        studentCommentsMap1.put(STUDENT_2_ID, STUDENT_2_COMMENT);
+
+        studentCommentsMap2 = new HashMap<>();
+        studentCommentsMap2.put(STUDENT_3_ID, STUDENT_3_COMMENT);
+        studentCommentsMap2.put(STUDENT_4_ID, STUDENT_4_COMMENT);
+
+        progress = new Progress();
+        progress.setId(PROGRESS_ID);
+        progress.setUrl(CANVAS_URL);
+        progress.setWorkflowState("queued");
+        progressOptional = Optional.of(progress);
+
+        Whitebox.setInternalState(assignmentSubmitter, "canvasAssignmentAssistant", canvasAssignmentAssistant);
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentService", assignmentService);
+        Whitebox.setInternalState(assignmentSubmitter, "sectionService", sectionService);
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceCourseService", attendanceCourseService);
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceService", attendanceService);
+        Whitebox.setInternalState(assignmentSubmitter, "canvasApiWrapperService", canvasApiWrapperService);
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
+
+
+
+    }
+
+    @Test
+    public void submitCourseAttendancesHappyPath() throws IOException, CanvasOutOfSyncException {
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_1_ID)).thenReturn(section1);
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_2_ID)).thenReturn(section2);
+        when(assignmentService.findBySection(section1)).thenReturn(attendanceAssignment1);
+        when(assignmentService.findBySection(section2)).thenReturn(attendanceAssignment2);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+        when(attendanceCourseService.findByCanvasCourseId(eq(COURSE_ID))).thenReturn(course);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_1_ID)).thenReturn(studentCommentsMap1);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_2_ID)).thenReturn(studentCommentsMap2);
+        when(canvasApiWrapperService.gradeMultipleSubmissionsBySection(any(), any())).thenReturn(progressOptional);
+
+        assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+        verify(canvasApiWrapperService, times(2)).gradeMultipleSubmissionsBySection(any(), any());
+    }
+
+    @Test
+    public void submitCourseAttendancesFailCanvasGradingError() throws IOException, CanvasOutOfSyncException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)